src/firefly/js/externalSource/__tests__/sprintf-test.js

```diff
@@ -0,0 +1,98 @@
+
+import {sprintf} from '../sprintf.js';
+
+
+describe('sprintf: ', () => {
+
+    test('standard:', () => {
+        expect(sprintf('%%'))       .toBe('%');
+        expect(sprintf('%b', 2))    .toBe('10');
+        expect(sprintf('%c', 65))	 .toBe('A');
+        expect(sprintf('%d', 2))    .toBe('2');
+        expect(sprintf('%i', 2))    .toBe('2');
+        expect(sprintf('%d', '2'))  .toBe('2');
+        expect(sprintf('%i', '2'))  .toBe('2');
+
+        expect(sprintf('%j', {foo: 'bar'}))	.toBe('{"foo":"bar"}');
+        expect(sprintf('%j', ['foo', 'bar']))	.toBe('["foo","bar"]');
+
+        expect(sprintf('%e', 2))    .toBe('2.000000e+0');
+        expect(sprintf('%.6e', 2))    .toBe('2.000000e+0');
```

Comment:
Does it support `%E` and `%G`?

Reply:
As far as sprintf is concern, no, we only support `%e` and `%g`.  This is probably because JavaScript's toExponential() prints it in `%e` format.  On the server-side, we support both. 
 On the client, I convert it to lowercase before formatting.

But now, since I've made quite a bit of changes to sprintf, It would be trivial to support uppercase E and G.  Do you think I should add it?

Reply:
It might make comparing with the original table easier (when you implement the original table format of IPAC tables from Gator).